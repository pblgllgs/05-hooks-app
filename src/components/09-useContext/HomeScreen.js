import React from 'react'

export const HomeScreen = () => {
    return (
        <>
            <h1>Home Screen</h1>
            <hr />
        </>
    )
}
