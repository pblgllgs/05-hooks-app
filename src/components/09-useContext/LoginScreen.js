import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const userlogin = {
        id:123,
        name:'pablo',
        email: 'pablo@gmail.com'
    } 

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className='btn btn-primary' 
                onClick={()=> setUser(userlogin)}
            >
                Login
            </button>
        </div>
    )
}
