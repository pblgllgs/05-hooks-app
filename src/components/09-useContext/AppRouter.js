import React from 'react'

import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <NavBar />
                <Switch>
                    <Route exact path="/" element={<HomeScreen />} />
                    <Route exact path="/about" element={<AboutScreen />} />
                    <Route exact path="/login" element={<LoginScreen />} />
                    <Route path="*" element={<HomeScreen />} />
                </Switch>
            </Router>
        </div>
    )
}