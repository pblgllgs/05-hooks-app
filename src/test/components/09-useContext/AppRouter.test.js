import { mount } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("pruebas en al <AppRouter />", () => {
    //se cre un user para el test
    const user = {
        id: 123,
        name: "pablo",
    };
    //capturamos el componente montado que sta dentro de un contexto que tiene un usuario
    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <AppRouter />
        </UserContext.Provider>
    );
    test("debe de mostrarse correctamente", () => {
        //se hace una comparacion con el snapshot
        expect(wrapper).toMatchSnapshot();
    });
});
