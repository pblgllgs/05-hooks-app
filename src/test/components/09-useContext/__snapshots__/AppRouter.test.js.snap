// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en al <AppRouter /> debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    Array [
      <h1>
        Home Screen
      </h1>,
      <hr />,
      <pre
        className="container"
      >
        {
   "id": 123,
   "name": "pablo"
}
      </pre>,
    ]
  </div>
</div>
`;
